File: ios/Porchivo/Views/AnnouncementsScreen.swift
//
//  AnnouncementsScreen.swift
//  Porchivo
//
//  Community tier — announcements feed fetched from org_announcements table.
//  Shows pinned + recent announcements with priority badges.
//

import SwiftUI

struct AnnouncementsScreen: View {
    @Environment(AppState.self) private var appState
    @Environment(\.porchivo) private var c

    var body: some View {
        ScrollView {
            VStack(spacing: 12) {
                if appState.announcements.isEmpty {
                    EmptyState(
                        symbol: "megaphone.fill",
                        title: "No announcements",
                        message: "Community announcements from your HOA or property management will appear here."
                    )
                } else {
                    ForEach(appState.announcements) { item in
                        AnnouncementCard(item: item)
                    }
                }
            }
            .padding(.horizontal, 16)
            .padding(.top, 16)
            .padding(.bottom, 24)
        }
        .background(c.background.ignoresSafeArea())
        .navigationTitle("Announcements")
        .navigationBarTitleDisplayMode(.inline)
    }
}

private struct AnnouncementCard: View {
    @Environment(\.porchivo) private var c
    let item: Announcement

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack(spacing: 8) {
                if item.isPinned {
                    Image(systemName: "pin.fill")
                        .font(.system(size: 11, weight: .bold))
                        .foregroundStyle(c.warmOrange)
                }
                Text(item.title)
                    .font(.system(size: 15, weight: .bold))
                    .foregroundStyle(c.textPrimary)
                    .lineLimit(2)
                Spacer()
                priorityPill
            }

            Text(item.body)
                .font(.system(size: 13))
                .foregroundStyle(c.textSecondary)
                .lineLimit(4)

            HStack(spacing: 6) {
                if let author = item.authorDisplayName, !author.isEmpty {
                    Text(author)
                        .font(.system(size: 11, weight: .medium))
                        .foregroundStyle(c.textMuted)
                    Text("·")
                        .font(.system(size: 11))
                        .foregroundStyle(c.textMuted)
                }
                Text(item.createdAt.formatted(date: .abbreviated, time: .shortened))
                    .font(.system(size: 11))
                    .foregroundStyle(c.textMuted)
            }
        }
        .padding(14)
        .background(c.surface, in: .rect(cornerRadius: Radius.lg))
        .shadow(color: c.textPrimary.opacity(0.05), radius: 6, y: 2)
    }

    private var priorityPill: some View {
        let tint: Color = item.priority == .urgent ? c.danger
            : (item.priority == .high ? c.warmOrange
            : (item.priority == .normal ? c.accent : c.textMuted))
        return Text(item.priority.label)
            .font(.system(size: 10, weight: .bold))
            .foregroundStyle(tint)
            .padding(.horizontal, 6)
            .padding(.vertical, 3)
            .background(tint.opacity(0.12), in: .rect(cornerRadius: Radius.sm))
    }
}

#Preview {
    NavigationStack {
        AnnouncementsScreen().environment(AppState())
    }
}

File: android/app/src/main/java/com/rork/porchivo/ui/screens/AnnouncementsScreen.kt
package com.rork.porchivo.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.outlined.Campaign
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import androidx.lifecycle.viewmodel.compose.viewModel
import androidx.navigation.NavController
import com.rork.porchivo.model.Announcement
import com.rork.porchivo.ui.components.EmptyState
import com.rork.porchivo.ui.theme.PorchivoTheme
import com.rork.porchivo.ui.viewmodel.AppViewModel
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

@Composable
fun AnnouncementsScreen(
    navController: NavController,
    modifier: Modifier = Modifier,
    appViewModel: AppViewModel = viewModel(),
) {
    val c = PorchivoTheme.colors
    val announcements by appViewModel.announcements.collectAsStateWithLifecycle()

    LazyColumn(
        modifier = modifier
            .fillMaxSize()
            .background(c.background),
        contentPadding = androidx.compose.foundation.layout.PaddingValues(
            start = 16.dp, end = 16.dp, top = 16.dp, bottom = 24.dp,
        ),
        verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        item {
            Row(verticalAlignment = Alignment.CenterVertically) {
                IconButton(onClick = { navController.popBackStack() }) {
                    Icon(
                        imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                        contentDescription = "Back",
                        tint = c.textPrimary,
                    )
                }
                Text(
                    text = "Announcements",
                    color = c.textPrimary,
                    fontSize = 20.sp,
                    fontWeight = FontWeight.Bold,
                )
            }
        }

        if (announcements.isEmpty()) {
            item {
                EmptyState(
                    icon = Icons.Outlined.Campaign,
                    title = "No announcements",
                    body = "Community announcements from your HOA or property management will appear here.",
                )
            }
        } else {
            items(announcements, key = { it.id }) { item ->
                AnnouncementCard(item = item)
            }
        }
    }
}

@Composable
private fun AnnouncementCard(item: Announcement, modifier: Modifier = Modifier) {
    val c = PorchivoTheme.colors
    val dateFormat = remember { SimpleDateFormat("MMM d, h:mm a", Locale.getDefault()) }

    Column(
        modifier = modifier
            .fillMaxWidth()
            .background(c.surface, RoundedCornerShape(16.dp))
            .padding(14.dp),
    ) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            if (item.isPinned) {
                Text(
                    text = "📌",
                    fontSize = 12.sp,
                )
            }
            Text(
                text = item.title,
                color = c.textPrimary,
                fontSize = 15.sp,
                fontWeight = FontWeight.Bold,
                modifier = Modifier.weight(1f),
                maxLines = 2,
            )
            PriorityPill(priority = item.priority)
        }

        Spacer(modifier = Modifier.height(8.dp))

        Text(
            text = item.body,
            color = c.textSecondary,
            fontSize = 13.sp,
            maxLines = 4,
        )

        Spacer(modifier = Modifier.height(6.dp))

        val authorText = item.authorDisplayName?.takeIf { it.isNotBlank() }
        val dateText = dateFormat.format(Date(item.createdAt))
        val metaText = if (authorText != null) "$authorText · $dateText" else dateText
        Text(
            text = metaText,
            color = c.textMuted,
            fontSize = 11.sp,
        )
    }
}

@Composable
private fun PriorityPill(priority: com.rork.porchivo.model.AnnouncementPriority) {
    val c = PorchivoTheme.colors
    val tint = when (priority) {
        com.rork.porchivo.model.AnnouncementPriority.URGENT -> c.danger
        com.rork.porchivo.model.AnnouncementPriority.HIGH -> c.warmOrange
        com.rork.porchivo.model.AnnouncementPriority.NORMAL -> c.accent
        com.rork.porchivo.model.AnnouncementPriority.LOW -> c.textMuted
    }
    Text(
        text = priority.label,
        color = tint,
        fontSize = 10.sp,
        fontWeight = FontWeight.Bold,
        modifier = Modifier
            .background(tint.copy(alpha = 0.12f), RoundedCornerShape(8.dp))
            .padding(horizontal = 6.dp, vertical = 3.dp),
    )
}
